<?xml version="1.0"?>
<launch>
    <arg name="config_file" default="lidar.yaml" />
    <arg name="topic_imu"           default="/zed2/zed_node/imu/data" />
    <arg name="path_bag"            default="/home/wz/dev-sdb/wz/Data/LVI-Calib/lvi_calib_318_2021-10-12-14-42-46.bag" />
    <arg name="bag_start"           default="1" />
    <arg name="bag_durr"            default="30" />
    <arg name="scan4map"            default="15" />
    <arg name="lidar_model"         default="VLP_16" />
    <arg name="ndtResolution"       default="0.5" /> <!-- 0.5 for indoor case and 1.0 for outdoor case -->

    <arg name="time_offset_padding" default="0.015" />
    <arg name="show_ui"    default="true" />

    <node pkg="lvi_calib" type="li_init_surfel_node" name="li_init_surfel_node" output="screen">
      <param name="config_file"  type="string" value="$(find lvi_calib)/cfg/$(arg config_file)" />
      <param name="topic_imu"         type="string"   value="$(arg topic_imu)" />
      <param name="topic_lidar"       type="string"   value="/rslidar_points" />
      <param name="LidarModel"        type="string"   value="$(arg lidar_model)" />
      <param name="path_bag"          type="string"   value="$(arg path_bag)" />
      <param name="bag_start"         type="double"   value="$(arg bag_start)" />
      <param name="bag_durr"          type="double"   value="$(arg bag_durr)" /> <!-- for data association -->
      <param name="scan4map"          type="double"   value="$(arg scan4map)" />
      <param name="ndtResolution"     type="double"   value="$(arg ndtResolution)" />

      <param name="time_offset_padding"   type="double"   value="$(arg time_offset_padding)" />
      <param name="show_ui"               type="bool"     value="$(arg show_ui)" />
    </node>

</launch>

